x/upgrade/keeper/keeper.go

```diff
@@ -34,7 +34,7 @@ type Keeper struct {
 	storeKey           storetypes.StoreKey             // key to access x/upgrade store
 	cdc                codec.BinaryCodec               // App-wide binary codec
 	upgradeHandlers    map[string]types.UpgradeHandler // map of plan name to upgrade handler
-	versionSetter      xp.ProtocolVersionSetter        // implements setting the protocol version field on BaseApp
+	baseapp            xp.ProtocolVersionSetter        // implements setting the protocol version field on BaseApp
```

Comment:
Why?

Reply:
Baseapp is passed here from app.go, the name version setter I found confusing here. Do you think it should be version setter?

Reply:
Yes.

Interfaces have the primary purpose of decoupling dependencies from the code which depends on them. The keeper depends on a ProtocolVersionSetter, which means anything that provides the method set of the ProtocolVersionSetter interface can be used to satisfy that dependency and "back" this field. It may be the case that the BaseApp is used to satisfy this dependency in "production" code, but that's incidental. The keeper doesn't (and shouldn't!) know about BaseApps. It just knows it needs to call something that can set a protocol version.

Reply:
happy revert, the interface is the same, but I see your point its confusing naming it baseapp

Reply:
Yeah, we should keep this as `versionSetter`.